_posts/2022-12-14-#26631.md

```diff
@@ -10,17 +10,117 @@ status: upcoming
 commit: 
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+### concepts
+
+- Each bitcoin transaction output carries a specific
+  [value](https://github.com/bitcoin/bitcoin/blob/6061eb6564105ad54703a7cf3282590d0e1a7f28/src/primitives/transaction.h#L159).
+  Bitcoin Core defines _dust_ as an output whose value is less than
+  what it would cost to spend this output.
+
+- An important goal of the Bitcoin network is to keep the resource costs
+  for running a fully-validating node to a minimum. One way to reduce
+  the storage requirement is to keep the size of the UTXO set small.
+
+- If it were possible for an attacker, or a careless user, to create
+  many tiny-value UTXOs, the UTXO set could be increased dramatically,
+  because whoever is able to spend these UTXOs (and thus remove them
+  from the UTXO set) would have little to no incentive to do so.
+
+- For this reason, Bitcoin Core has a policy of not accepting into its
+  mempool or relaying any transaction with a _dust_ output, that is, an
+  output whose value is below the _dust_ limit.
+
+### details
+
+- When validating an incoming transaction, policy code calculates the
+  fee, at a particular feerate, to "pay for" both the output and the (later)
+  spending input. This fee is proportional to the sum of the sizes,
+  measured in virtual bytes, of both the input and output.
+
+- If the output value is below this (hypothetical)
+  fee, it is considered
+  [_dust_](https://github.com/bitcoin/bitcoin/blob/1ea02791f3d81c7716d9ea455971203f74d7a107/src/policy/policy.cpp#L65);
+  it would cost more to spend this output than its value.
+
+- An output's virtual size is just it's physical size, but an input typically
+  also includes _witness data_ which is discounted: 4 bytes of witness data
+  is counted as one byte of virtual data.
+
+- Rather than the dust feerate being hardcoded, `bitcoind` includes a
+  configuration option `-dustrelayfee=<feerate>` to set this value.
+
+- This feerate is in units of BTC per kvB (1000 virtual bytes).
+  The default is 0.00003 BTC per kvB (3000 sats/kvB or 3 sat/vB).
+
+- More information can be found on
+  [stackexchange](https://bitcoin.stackexchange.com/questions/10986/what-is-meant-by-bitcoin-dust).
+
+- There are several kinds of outputs (for example, P2PK, P2PKH), and each has
```

Comment:
There are infinite outputs; we enumerate the *standard* ones. We cannot predict the size of a P2SH input without knowing the script.
```suggestion
- There are several kinds of standard outputs (for example, P2PK, P2PKH), and some have
```